### PR TITLE
page-object files have been modified to improve locators.

### PR DESCRIPTION
**Pull request by JuanImbachi**
The explanation of each change appears in the code as a comment.
image
image

### CODE REVIEW

File: src/page/address-step.page.ts [hunk 1]
```diff
@@ -4,7 +4,9 @@ export class AddressStep {
   private checkoutBtn: ElementFinder;
 
   constructor () {
-    this.checkoutBtn = $('#center_column > form > p > button > span');
+   // this.checkoutBtn = $('#center_column > form > p > button > span');
+   // The name of the element is unique in the view, so it makes easier to identify the button.
+    this.checkoutBtn = $('[name="processAddress"]');
```
Comment (holgiosalos) [inline src/page/address-step.page.ts:9]: 👍

File: src/page/shipping-step.page.ts [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
   private checkoutBtn: ElementFinder;
 
   constructor () {
-    this.checkCgv = $('#cgv');
-    this.checkoutBtn = $('#form > p > button > span');
+    // This selector is the same because is already using the ID
+    this.acceptTermsCheck = $('#cgv');
+    // this.checkoutBtn = $('#form > p > button > span');
+    // This selector changed because  whit the name of the button is enough to find the element.
+    this.checkoutBtn = $('[name="processCarrier"]');
```
Comment (holgiosalos) [inline src/page/shipping-step.page.ts:12]: 👍

File: src/page/product-added-modal.page.ts [hunk 3]
```diff
@@ -4,7 +4,11 @@ export class ProductAddedModalPage {
   private addToCart: ElementFinder;
 
   constructor () {
-    this.addToCart = $('[style*="display: block;"] .button-container > a');
+    // this.addToCart = $('[style*="display: block;"] .button-container > a');
+    // I preferred to use the id of a div instead of use the value of a property like 'style'.
+    // I consider is easier to identify the element with the ID of the div and the class name
+    // where the button is.
+    this.addToCart = $('#layer_cart .button-container');
```
Comment (holgiosalos) [inline src/page/product-added-modal.page.ts]: 👍

**Review by holgiosalos — COMMENTED** [review]
I will finish your review tomorrow.

File: src/page/menu-content.page.ts [hunk 4]
```diff
@@ -4,7 +4,10 @@ export class MenuContentPage {
   private tShirtMenu: ElementFinder;
 
   constructor () {
-    this.tShirtMenu = $('#block_top_menu > ul > li:nth-child(3) > a');
+    // this.tShirtMenu = $('#block_top_menu > ul > li:nth-child(3) > a');
+    // Each type 'a' element in this block of code has a specific value for the title property,
+    // so I changed the selector to use the property like it was an ID.
+    this.tShirtMenu = $('#block_top_menu > ul > li > a[title="T-shirts"]');
```
Comment (holgiosalos) [inline src/page/menu-content.page.ts:10]: 👍

File: src/page/product-list.page.ts [hunk 5]
```diff
@@ -4,7 +4,9 @@ export class ProductListPage {
   private productList: ElementFinder;
 
   constructor () {
-    this.productList = $('#center_column a.button.ajax_add_to_cart_button.btn.btn-default');
+  // The class of the element we want to click on is so specific
+  // that we can use it to identify the element on the whole page.
+    this.productList = $('a.button.ajax_add_to_cart_button.btn.btn-default');
```
Comment (holgiosalos) [inline src/page/product-list.page.ts:9]: 👍

File: src/page/product-added-modal.page.ts (hunk 3 above)
Comment (holgiosalos) [reply src/page/product-added-modal.page.ts]: I have changed of opinion: If you leave just `#layer_cart .button-container`  you are pointing to two possible buttons: proceed to checkout and continue shopping. So, you will need to specify and additional element in the current locator or find another locator.

image

File: src/page/sign-in-step.page.ts [hunk 6]
```diff
@@ -6,7 +6,10 @@ export class SignInStep {
   private passwordField: ElementFinder;
 
   constructor () {
-    this.signInBtn = $('#SubmitLogin > span');
+    // this.signInBtn = $('#SubmitLogin > span');
+    // I will only modify the first selector because with the ID is enough.
+    // The email and password selectors will not change because they are already using their IDs.
+    this.signInBtn = $('#SubmitLogin');
```
Comment (holgiosalos) [inline src/page/sign-in-step.page.ts:12]: 👍

File: src/page/payment-step.page.ts [hunk 7]
```diff
@@ -4,7 +4,10 @@ export class PaymentStep {
   private paymentOpt: ElementFinder;
 
   constructor () {
-    this.paymentOpt = $('#HOOK_PAYMENT > div:nth-child(1) > div > p > a');
+   // this.paymentOpt = $('[#HOOK_PAYMENT > div:nth-child(1) > div > p > a]');
+   // This selector changed because whit the value of the title property
+   // is enough to find the element.
+    this.paymentOpt = $('[title="Pay by bank wire"]');
```
Comment (holgiosalos) [inline src/page/payment-step.page.ts:10]: 👍

File: src/page/bank-payment.page.ts [hunk 8]
```diff
@@ -4,7 +4,10 @@ export class BankPaymentPage {
   private confirmOrder: ElementFinder;
 
   constructor () {
-    this.confirmOrder = $('#cart_navigation > button > span');
+    // this.confirmOrder = $('#cart_navigation > button > span');
+    // I changed the selector because in the code block of #cart_navigation
+    // is only one 'span', so whit those two thing we can identidy the element that we need.
+    this.confirmOrder = $('#cart_navigation span');
```
Comment (holgiosalos) [inline src/page/bank-payment.page.ts]: I think that this doesn't make any difference with the original one, you can play a little bit more, try to use a different locator, for example, the text related with the button seems to be unique "I confirm my order". Find a locator in where you can use that text.

File: src/page/order-summary.page.ts [hunk 9]
```diff
@@ -4,7 +4,10 @@ export class OrderSummaryPage {
   private expectedElement: ElementFinder;
 
   constructor () {
-    this.expectedElement = $('#center_column > div > p > strong');
+   // this.expectedElement = $('#center_column > div > p > strong');
+   // The class where the title appear is unique in the view, so with it we can identify
+   // the element easily.
+    this.expectedElement = $('.cheque-indent');
```
Comment (holgiosalos) [inline src/page/order-summary.page.ts:10]: 👍

File: src/page/bank-payment.page.ts (hunk 8 above)
Comment (JuanImbachi) [reply src/page/bank-payment.page.ts]: until the moment, i have not find yet a locator that works, i will keep trying.

**Comment by yesid-lopez** [comment]
Do not forget to delete the commented lines in your code, you can add them with GitHub instead

File: src/page/bank-payment.page.ts (hunk 8 above)
Comment (yesid-lopez) [reply src/page/bank-payment.page.ts]: what about using the protractor functions instead?

File: src/page/bank-payment.page.ts [hunk 10]
```diff
@@ -7,7 +7,7 @@ export class BankPaymentPage {
     // this.confirmOrder = $('#cart_navigation > button > span');
     // I changed the selector because in the code block of #cart_navigation
     // is only one 'span', so whit those two thing we can identidy the element that we need.
-    this.confirmOrder = $('#cart_navigation span');
+    this.confirmOrder = element(by.cssContainingText('span', 'I confirm my order'));
```
Comment (holgiosalos) [inline src/page/bank-payment.page.ts:10]: good job! also, other alternative that you could use was: `element(by.partialButtonText('I confirm my order'))`. Both options are fine :)

**Review by holgiosalos — APPROVED** [review]

**Pull request merged**